### ISSUE TITLE
[Impoorvment] MediaListener and Change detection

### ISSUE THREAD

**Issue by shlomiassaf** [justindujardin/ng2-material#228]
Hi,

Currently, to implement a responsive UI for `SideNav` one must use the `Media` service and listen to media query changes.

When a media query event fires the current example uses `ApplicationRef.tick()`, this is done to trigger change detection ([code](https://github.com/justindujardin/ng2-material/blob/master/examples/app.ts#L61))

While this solves the problem, it requires manual action by the user.

The underlying problem is that events fired from `MediaQueryList` are not in the Zone, i.e: Zone.js does not wrap such events thus it will never trigger a change detection check like a click or setTimeout will.
See [code](https://github.com/justindujardin/ng2-material/blob/master/ng2-material/core/util/media.ts#L65-L66)

To avoid users having to apply `ApplitcationRef.tick()` we can use `NgZone` to do it automatically:

In the constructor of the class [MediaListener](https://github.com/justindujardin/ng2-material/blob/master/ng2-material/core/util/media.ts#L65), instead of:
```ts
    this._listener = (mql: MediaQueryList) => this.onMatched.next(mql);
```

do:
```ts
    this._listener = (mql: MediaQueryList) => _media.ngZone.run(() => this.onMatched.next(mql));
```

The `Media` class will need a little adjustment:
```ts

/**
 * Injectable class for being notified of changes to viewport media queries.
 */
@Injectable()
export class Media {
  private _cache: {[query: string]: IMediaQueryCache} = {};

  constructor(public viewport: ViewportHelper, public ngZone: NgZone) {

  }
  ...
```

I would post a PR but I believe it's not a smart thing to do given the changes done in **dev** for RC1.
I also don't have the project cloned and it will take a lot of time for such a small change.

**Comment by ollwenjones** [comment]
Thanks for the excellent explanation. I'm sure a PR would get queued with the other changes appropriately, but also pretty sure one of us will pick this up.

**Comment by justindujardin** [comment]
Thanks for reporting, closed with `0.4.1` and https://github.com/justindujardin/ng2-material/commit/b7385e1d258296cbf0fc8d544d1aa9958856f58e
